kola/tests/kubernetes/basic.go

```diff
@@ -20,10 +20,58 @@ import (
 	"strings"
 	"time"
 
+	"github.com/coreos/mantle/kola/register"
 	"github.com/coreos/mantle/platform"
 	"github.com/coreos/mantle/util"
 )
 
+// register a separate test for each version tag
+var basicTags = []string{
+	"v1.1.7_coreos.2",
+	"v1.1.8_coreos.0",
+}
+
+func init() {
+	for i := range basicTags {
+		// use closure to store a version tag in a Test
+		t := basicTags[i]
+		f := func(c platform.TestCluster) error {
+			return CoreOSBasic(c, t)
+		}
+
+		register.Register(&register.Test{
+			Name:        "google.kubernetes.basic." + t,
+			Run:         f,
+			ClusterSize: 0,
+			Platforms:   []string{"gce", "aws"},
+		})
+	}
+}
+
+// Run basic smoke tests on cluster. Assumes master is machine index 1,
+// workers make up the rest.
+func CoreOSBasic(c platform.TestCluster, version string) error {
+	if err := setupCluster(c, 2, version); err != nil {
+		return err
+	}
+
+	master := c.Machines()[1]
```

Comment:
Oh dang, this isn't even safe! I was having tons of trouble testing some of this because I would manhole into c.Machines()[1] but when I ran it without manhole it was logging into the wrong machine.

We can't count on the order of the machine list because its actually stored in `basecluster.machmap`. I didn't realize or had forgot that detail. Ugh.